mmrazor/models/algorithms/quantization/mm_architecture.py

```diff
@@ -54,21 +57,37 @@ def __init__(self,
                  init_cfg=None):
 
         if data_preprocessor is None:
-            data_preprocessor = {}
-        # The build process is in MMEngine, so we need to add scope here.
-        data_preprocessor.setdefault('type', 'mmcls.ClsDataPreprocessor')
+            data_preprocessor = getattr(architecture, 'data_preprocessor',
+                                        dict())
         super().__init__(architecture, data_preprocessor, init_cfg)
-        if float_checkpoint:
-            _ = load_checkpoint(self.architecture, float_checkpoint)
-            self.architecture._is_init = True
 
         self.quantizer = MODELS.build(quantizer)
         self.input_shapes = input_shapes
         self.forward_modes = forward_modes
 
-        self.qmodels = self._build_qmodels(self.architecture)
+        # Replace syncbn and _BatchNormXd (in mmengine) with batchnorm2d
+        self.quantizer.convert_batchnorm2d(self.architecture)
+
+        if float_checkpoint:
+            _ = load_checkpoint(self.architecture, float_checkpoint)
+            self.architecture._is_init = True
 
-        self.sync_qparams('predict')
+        self.qmodels = self._build_qmodels(self.architecture)
+        self.sync_qparams('tensor')
```

Comment:
had better use `predict` as default, `predict` mode is required, but `tensor` mode is optional

Reply:
`data_samples` can not be `None` in detectors during prediction. But we need to make the dummy prediction in `_build_qmodels`. It is more convenient to use tensor mode.